config/routes.rb

```diff
@@ -7,5 +7,7 @@
   resources :games, only: %i[create new show], param: :slug do
     resources :players, only: %i[create new]
   end
+
+  post '/games/leave_game', to: 'games#leave_game'
```

Comment:
It would be nice to have the slug as part of the route if possible.

`post '/games/(/:slug)/players/(/:id)/leave` would match the established pattern: see '[nested resources](https://guides.rubyonrails.org/routing.html#nested-resources)'.

So:

```ruby
resources :games, only: %i[create new show], param: :slug do
  resources :players, only: %i[create new] do
    post '/games/leave_game', to: 'games#leave_game'
  end
end
```